src/PaymentPart/Output/AbstractOutput.php

```diff
@@ -77,6 +79,18 @@ protected function getInformationElements(): array
             $informationElements[] = Text::create($this->qrBill->getPaymentReference()->getFormattedReference());
         }
 
+        if ($this->qrBill->getPaymentAmountInformation()->getAmount() === (float)0) {
+            $doNotUseForPaymentText = Translation::get('doNotUseForPayment', $this->language);
+            $additionalInformation = $this->qrBill->getAdditionalInformation();
+            if (isset($additionalInformation)) {
+                $text = $doNotUseForPaymentText."\n".$additionalInformation->getMessage();
+            } else {
+                $text = $doNotUseForPaymentText;
+            }
+            $newAdditionalInformation = AdditionalInformation::create($text);
+            $this->qrBill->setAdditionalInformation($newAdditionalInformation);
+        }
```

Comment:
This is the wrong place for this functionality.

If you scan the example code [here](https://www.paymentstandards.ch/dam/downloads/notification-en.pdf) (page 7), you can see that the "do not use for payment" message also appears in the qr code itself, not just on the payment part.

This will require some refactoring, as currently generating just the qr code is unaware of the language. I will have a look at this.